lxd/network/driver_bridge.go

```diff
@@ -388,10 +388,23 @@ func (n *bridge) Validate(config map[string]string) error {
 			allowedNets = append(allowedNets, dhcpSubnet)
 		}
 
-		_, err := parseIPRanges(config["ipv4.ovn.ranges"], allowedNets...)
+		ovnRanges, err := parseIPRanges(config["ipv4.ovn.ranges"], allowedNets...)
 		if err != nil {
 			return errors.Wrapf(err, "Failed parsing ipv4.ovn.ranges")
 		}
+
+		dhcpRanges, err := parseIPRanges(config["ipv4.dhcp.ranges"], allowedNets...)
+		if err != nil {
+			return errors.Wrapf(err, "Failed parsing ipv4.dhcp.ranges")
+		}
+		for _, ovnRange := range ovnRanges {
+			for _, dhcpRange := range dhcpRanges {
+				if IPRangesOverlap(ovnRange, dhcpRange) {
+					return fmt.Errorf("The range specified in 'ipv4.ovn.ranges' cannot overlap with 'ipv4.dhcp.ranges'")
```

Comment:
It would be useful to the user to show them the offending range in the error (have also modified the quoting of fields to use double quotes).

E.g.

```
return fmt.Errorf(`The range specified in "ipv4.ovn.ranges" (%q) cannot overlap with "ipv4.dhcp.ranges"`, ovnRange)
```

I think this will require you adding a `String()` function to the IPRange struct type so that `fmt` can render it.